File: CodingSamples/Java/Platform-Enterprise/TCPSocktetTest/demo2/src/main/java/app/ItemInfo.java
package app;

public record ItemInfo(double price, int stock) {
    
    public static ItemInfo parse(String message){
        String[] parts = message.split(",");
        double price = Double.parseDouble(parts[0].substring(6));
        int stock = Integer.parseInt(parts[1].substring(6));
        return new ItemInfo(price, stock);
    }
}

File: CodingSamples/DotNet/Platform-Enterprise/TcpSocketTest/DemoApp3/ItemInfo.cs
namespace DemoApp;

readonly record struct ItemInfo(double Price, int Stock)
{
    public static ItemInfo Parse(string message)
    {
        string[] segments = message.Split(',');
        double price = double.Parse(segments[0][6 ..]);
        int stock = int.Parse(segments[1][6 ..]);
        return new ItemInfo(price, stock);
    }

}